stan/math/prim/functor/integrate_ode_rk45.hpp

```diff
@@ -120,26 +121,34 @@ std::vector<std::vector<return_type_t<T1, T2, T_t0, T_ts>>> integrate_ode_rk45(
 
   // avoid recording of the initial state which is included by the
   // conventions of odeint in the output
+  size_t timestep = 0;
   auto filtered_observer
       = [&](const std::vector<double>& coupled_state, double t) -> void {
     if (!observer_initial_recorded) {
       observer_initial_recorded = true;
       return;
     }
     observer(coupled_state, t);
+    timestep++;
   };
 
   // the coupled system creates the coupled initial state
   std::vector<double> initial_coupled_state = coupled_system.initial_state();
 
   const double step_size = 0.1;
-  integrate_times(
-      make_dense_output(absolute_tolerance, relative_tolerance,
-                        runge_kutta_dopri5<std::vector<double>, double,
-                                           std::vector<double>, double>()),
-      std::ref(coupled_system), initial_coupled_state, std::begin(ts_vec),
-      std::end(ts_vec), step_size, filtered_observer,
-      max_step_checker(max_num_steps));
+  try {
+    integrate_times(
+        make_dense_output(absolute_tolerance, relative_tolerance,
+                          runge_kutta_dopri5<std::vector<double>, double,
+                                             std::vector<double>, double>()),
+        std::ref(coupled_system), initial_coupled_state, std::begin(ts_vec),
+        std::end(ts_vec), step_size, filtered_observer,
+        max_step_checker(max_num_steps));
+  } catch (no_progress_error& e) {
+    throw_domain_error("integrate_ode_rk45", "", ts_vec[timestep + 1],
```

Comment:
Why a mix of explicit calls to `throw` and this `throw_domain_error` helper?

Reply:
It appears most of this code was written with throw instead of using the internal error functions.

Where I added new errors I used the internal error functions. Didn't want to do the code cleanup everywhere.

Reply:
Change "most" to "some"

Reply:
I changed a couple of the manual throws to `throw_domain_errors`. I kindof abused the syntax to get a reasonable error message, but they're tested so if the formatting changes someone will at least have to look at them.